Katello/katello
# Fixes #33608 - Use settings DSL

#### What are the changes introduced in this pull request?

* Use the new Settings DSL to define Katello settings
* Before: Settings were defined in `content.rb` and loaded from `engine.rb`. Now: Settings are defined in `plugin.rb` directly. 
* Move active record callbacks and actions to setting_extensions
* Move our one setting validator to `app/lib/katello/validators`
* Move setting helper methods to `settings` block of `plugin.rb`
* clean up some capitalization etc.
* hopefully not break anything

#### Considerations taken when implementing this change?

see the inline comments I will add

#### What are the testing steps for this pull request?

* Navigate to Administer > Settings and take note of some Content setting values, particularly non-default ones (or set some, if you like)
* Stop your server; Run the migration; Start your server (Warning: This migration is not reversible)
* Visit Administer > Settings and verify that all Content settings are still present and still have their values
* Verify that you can change setting values without error

* Test the validator: try doing `Setting[:content_default_http_proxy] = 'foo'`; you should get `ActiveRecord::RecordInvalid: Validation failed: There is no such HTTP proxy`
* Test the after-action: Set that same setting to a valid value (nil is also valid) and verify that `SettingExtensions#update_global_proxies` runs `::Actions::BulkAction` and `::Actions::Katello::Repository::UpdateHttpProxyDetails`
* Do a `rake katello:reset` and verify that all the migrations can complete without error

__Important:__ After switching away from this branch you'll probably need to rebase on this PR or things will start breaking.
